src/types.rs

```diff
@@ -407,45 +413,36 @@ impl<'buf> RecvMsgOut<'buf> {
                 .read_unaligned()
         };
 
-        let msghdr_name_len = msghdr.msg_namelen as _;
-        let msghdr_control_len = msghdr.msg_controllen as _;
-
-        // Check total length upfront, so that further logic here
-        // below can safely use unchecked/saturating math.
```

Comment:
Would it be possible to bring back the saturating math in all the start/end computations below? I'm not sure why you changed them, but that doesn't seem to be part of the actual fix.

Reply:
They're unnecessary. We check the lengths above and use `min` everywhere, so it's impossible to overflow.

Reply:
Sidenote: I filed https://github.com/rust-lang/rust/issues/120433 because we're not getting bounds check ellison which is annoying.

Reply:
Yes we agree that they are all properly bound. That's why it seems reasonable to let the compiler know we aren't interested in panic points or overflowing math in there. Anyway, fine either ways.

Reply:
Oh I see, the saturating version does generate slightly better assembly with one less branch: https://rust.godbolt.org/z/Pdj3fxsnn. Through really I'd like to see LLVM fixed.

In the meantime, do you want a PR with saturating_add? Or we could also just tell LLVM stuff is in bound and then you get a completely branchless version:
```rust
if ..._end > buffer.len() {
  unsafe { std::hint::unreachable_unchecked(); }
}
```

Reply:
No need to, what we merged is also fine as is.

Reply:
Ok, thanks!